File: src/easy/FinalPricesWithDiscount.swift
func finalPricesWithDiscount(_ prices: [Int]) -> [Int] {
  var result = Array(prices)
  for i in 0..<result.count {
    for j in i + 1..<result.count {
      if prices[j] <= prices[i] {
        result[i] = prices[i] - prices[j]
        break
      }
    }
  }
  return result
}

File: src/main/php/Easy/FinalPricesWithDiscount.php
<?php

declare(strict_types=1);

namespace Src\Easy;

class FinalPricesWithDiscount
{
    /**
     * @param  int[]  $prices
     * @return int[]
     */
    public function solve(array $prices): array
    {
        $result = [...$prices];
        for ($i = 0; $i < count($prices); $i++) {
            for ($j = $i + 1; $j < count($prices); $j++) {
                if ($prices[$j] <= $prices[$i]) {
                    $result[$i] = $prices[$i] - $prices[$j];
                    break;
                }
            }
        }

        return $result;
    }
}
